### PR TITLE
Add custom Event Category & custom Tag

### PR DESCRIPTION
**Pull request by ghost**
Adds ability to specify which Event Category (instead of default "Generated") and/or Tag (instead of default "Automated") will be added to the generated events.

Screencast: default behavior > http://p.tri.be/34kBat
Screencast: custom category and tag > http://p.tri.be/yP2afM

### CODE REVIEW

File: src/Tribe/Generator/Event.php [hunk 1]
```diff
@@ -118,6 +120,42 @@ public function random_event_data( $from_date, $to_date, $is_featured, $is_virtu
...
 		];
 
+		if( $has_category ) {
+			$custom_category_term = wp_insert_term( $event_category, 'tribe_events_cat' );
+			$custom_category_id = $custom_category_term instanceof \WP_Error
+				? (int)$custom_category_term->get_error_data()
+				: $custom_category_term['term_id'];
+			$random_event_data = array_merge( $random_event_data, [
+				'category'            => [ $custom_category_id ]
+			] );
```
Comment (bordoni) [inline src/Tribe/Generator/Event.php]:
```suggestion

			if ( $custom_category_term instanceof \WP_Error ) {
				$custom_category_id = (int) $custom_category_term->get_error_data();
			} else {
				$custom_category_id = $custom_category_term['term_id'];
			}
				
			$random_event_data = array_merge(
				$random_event_data,
				[
					'category' => [ $custom_category_id ]
				]
			);
```

File: src/Tribe/Generator/Event.php [hunk 2]
```diff
@@ -118,6 +120,42 @@ public function random_event_data( $from_date, $to_date, $is_featured, $is_virtu
...
+		} else {
+			$category_term = wp_insert_term( 'Generated', 'tribe_events_cat' );
+			$category_id = $category_term instanceof \WP_Error
+				? (int)$category_term->get_error_data()
+				: $category_term['term_id'];
+			$random_event_data = array_merge( $random_event_data, [
+				'category'            => [ $category_id ]
+			] );
```
Comment (bordoni) [inline src/Tribe/Generator/Event.php]:
```suggestion
			$category_term = wp_insert_term( 'Generated', 'tribe_events_cat' );

			if ( $category_term instanceof \WP_Error ) {
				$category_id = (int) $category_term->get_error_data();
			} else {
				$category_id = $category_term['term_id'];
			}
				
			$random_event_data = array_merge(
				$random_event_data,
				[
					'category' => [ $category_id ]
				]
			);
```

File: src/Tribe/Generator/Event.php [hunk 3]
```diff
@@ -118,6 +120,42 @@ public function random_event_data( $from_date, $to_date, $is_featured, $is_virtu
...
+		}
+
+		if( $has_tag ) {
+			$custom_tag_term = wp_insert_term( $event_tag, 'post_tag', [ 'slug' => $event_tag ] );
+			$custom_tag_id = $custom_tag_term instanceof \WP_Error
+				? (int)$custom_tag_term->get_error_data()
+				: $custom_tag_term['term_id'];
+			$random_event_data = array_merge( $random_event_data, [
+				'tag'                 => [ $custom_tag_id ],
+			] );
```
Comment (bordoni) [inline src/Tribe/Generator/Event.php]:
```suggestion
			$custom_tag_term = wp_insert_term( $event_tag, 'post_tag', [ 'slug' => $event_tag ] );

			if ( $custom_tag_term instanceof \WP_Error ) {
				$custom_tag_id = (int) $custom_tag_term->get_error_data();
			} else {
				$custom_tag_id = $custom_tag_term['term_id'];
			}
				
			$random_event_data = array_merge(
				$random_event_data,
				[
					'tag' => [ $custom_tag_id ]
				]
			);
```

File: src/Tribe/Generator/Event.php [hunk 4]
```diff
@@ -118,6 +120,42 @@ public function random_event_data( $from_date, $to_date, $is_featured, $is_virtu
...
+		} else {
+			$tag_term = wp_insert_term( 'Automated', 'post_tag', [ 'slug' => 'automated-tdgext' ] );
+			$tag_id = $tag_term instanceof \WP_Error
+				? (int)$tag_term->get_error_data()
+				: $tag_term['term_id'];
+			$random_event_data = array_merge( $random_event_data, [
+				'tag'                 => [ $tag_id ],
+			] );
```
Comment (bordoni) [inline src/Tribe/Generator/Event.php]:
```suggestion
			$tag_term = wp_insert_term( 'Automated', 'post_tag', [ 'slug' => 'automated-tdgext' ] );
			
			if ( $tag_term instanceof \WP_Error ) {
				$tag_id = (int) $tag_term->get_error_data();
			} else {
				$tag_id = $tag_term['term_id'];
			}
				
			$random_event_data = array_merge(
				$random_event_data,
				[
					'tag' => [ $tag_id ]
				]
			);
```

File: src/Tribe/Generator/Event.php [hunk 5]
```diff
@@ -168,45 +206,45 @@ public function random_event_data( $from_date, $to_date, $is_featured, $is_virtu
...
+			];
+
+			if ( 'Yearly' === $type )  {
+				$month = $recurrence_date->format('n');
+				foreach ( $default_data['recurrence']['rules'] as $index => $rule ) {
+					$rule['custom']['year'] = [
+						'month'    => [ $month ],
+						'same-day' => 'yes'
+					];
+					$default_data['recurrence']['rules'][$index] = $rule;
```
Comment (bordoni) [inline src/Tribe/Generator/Event.php]:
```suggestion
					$default_data['recurrence']['rules'][ $index ] = $rule;
```

File: src/Tribe/Generator/Event.php [hunk 6]
```diff
@@ -168,45 +206,45 @@ public function random_event_data( $from_date, $to_date, $is_featured, $is_virtu
...
+				}
+			}
+
+			if ( 'Weekly' === $type ) {
+				$weekday = (int) $recurrence_date->format('w');
+				foreach ( $default_data['recurrence']['rules'] as $index => $rule ) {
+					$rule['custom']['week'] = [ 'day' => [ $weekday ] ];
+					$default_data['recurrence']['rules'][$index] = $rule;
```
Comment (bordoni) [inline src/Tribe/Generator/Event.php]:
```suggestion
					$default_data['recurrence']['rules'][ $index ] = $rule;
```

**Review by bordoni — CHANGES_REQUESTED** [review]
Small changes.

**Comment by ghost** [comment]
All good!

**Pull request merged**